os/fs/mqueue/Make.defs

```diff
@@ -53,9 +53,16 @@
 # Include POSIX message queue support
 
 ifneq ($(CONFIG_DISABLE_MQUEUE),y)
```

Comment:
1. Mqueue (signal, env, pthread and timer as well) is supported from os and libc folders.
    CONFIG_DISABLE_MQUEUE means we can't use some mqueue functions which are at os.
    But by your commit (this commit), it is allowed.
2. Also it means POSIX compatible APIs. But yours aren't.

By above, we had better make a new config to say it is not POSIX-compatible but it works.
This commit checks two configs, NFILE_DES and DISABLE_MQUEUE.
```
if (NFILE == 0) && !defined(DISABLE_MQ)
```
When above is true, our POSIX-incompatitble mq apis are working.
It's good to reduce memory usages. But we must notice it is POSIX-incompatible

Reply:
Hi Sunghan, 
Our alternative mq implementation is mainly to provide mq even without vfs support. 
With this alternative implementation, all mq API's provided in the library will work as before. 
Only mq_notify() will be an exception. It will not work since it depends on Signals. Hence mq_notify is disabled in this case. The remaining mq API's will still remain Posix compliant. 

Also, in the minimal profile, most of the posix functionality is disabled. For example, VFS, Signals, etc.. So, when selecting minimal defconfig, user cannot assume Posix compliance. Pls Let me know your opinion.